### PR TITLE
Push Notifications: Signup and Kudos data

### LINKED ISSUE #177: Push Notifications: Add data to signup push
We need to add a JSON object to each notification that the app can parse to figure out where to send the user next. When a user gets a notification that someone in their group signed up for a campaign, we need to attach the group info to the notification. 

```
{
    "data": {
        "campaign_id": "362",
        "users": [
            {
                "_id": "5430e850dt8hbc541c37tt3d",
                "email": "test@dosomething.org",
                "birthdate": "12/17/91",
                "first_name": "First",
                "last_name": "Last",
                "updated_at": "2015-06-03 22:07:07",
                "created_at": "2015-06-03 22:07:07",
                "campaigns": [
                    {
                        "drupal_id": "362",
                        "signup_id": 124,
                        "signup_source": "android",
                        "updated_at": "2015-06-04 17:48:17",
                        "created_at": "2015-06-04 17:48:17"
                    }
                ]
            },
            {
                "_id": "5480c950bffebc651c8b456f",
                "email": "test1@dosomething.org",
                "mobile": "5555550101",
                "birthdate": "12/17/91",
                "first_name": "First",
                "last_name": "Last",
                "campaigns": [
                    {
                        "drupal_id": "362",
                        "signup_id": 130,
                        "signup_source": "iOS",
                        "signup_group": 124
                    }
                ],
                "updated_at": "2015-06-03 22:07:07",
                "created_at": "2015-06-03 22:07:07"
            }
        ]
    }
}
```

### PR DESCRIPTION
**Pull request by sbsmith86**
#### What's this PR do?
* Adds custom data to the kudos and sign up push notification data.
* Adds two new unit test to check that the notification data is built correctly.

*minor changes*
- Adds a new `reportbackItemContent` function that retrieves a specific reportback item from drupal.
- Moves the event firing for the kudos event into the `storeKudos` function and instead of taking a user as parameter, takes the reportback item that received the kudo. 

#### Where should the reviewer start?
- `SendKudoPushNotifiacation` and ` SendSignupPushNotification` this is where the custom data object gets built out. 

#### How should this be manually tested?
Run the unit tests!
Kudos: `vendor/bin/phpunit --group kudoPushNotification`
Signup: `vendor/bin/phpunit --group signupPushNotification`

#### What are the relevant tickets?
Closes #177 and #176

### CODE REVIEW

File: app/Services/DrupalAPI.php [hunk 1]
```diff
@@ -249,6 +249,9 @@ public function storeKudos($drupal_id, $request)
 
         $body = $response->json();
 
+        // Fire kudo event.
+        event(new UserGotKudo($request->reportback_item_id));
```
Comment (jonuy) [inline app/Services/DrupalAPI.php]: I think it'd be better to keep this event in the KudosController. It decouples the generic Drupal services logic from event triggering that's more specific to northstar.

I guess I'm treating Services/DrupalAPI as if it were a library that we could open source and distribute as a library in itself.

File: app/Handlers/Events/SendKudoPushNotification.php [hunk 2]
```diff
@@ -2,43 +2,92 @@
...
+
+    /**
+     * Compile push data per user to send to Parse.
+     *
+     * @param UserGotKudo $event
+     * @return array
+     */
+    public function createPushData(UserGotKudo $event)
+    {
+
```
Comment (jonuy) [inline app/Handlers/Events/SendKudoPushNotification.php]: :space_invader:

File: app/Handlers/Events/SendSignupPushNotification.php [hunk 3]
```diff
@@ -2,53 +2,108 @@
...
+                    'extras' => [
+                        'invitation' => [
+                            'message' => $message,
+                            'group' => [
+                                'data' => [
+                                    'campaign_id' => $event->campaign->drupal_id,
+                                    'users' => $group,
+                                ],
+                            ],
+                        ]
```
Comment (jonuy) [inline app/Handlers/Events/SendSignupPushNotification.php]: `,`

**Comment by jonuy** [comment]
Tests passing. Minor code style things. And just the one comment on where you're firing the event. :+1:

**Comment by sbsmith86** [comment]
@jonuy thank for the review! Made the updates :) going to merge.

**Pull request merged**